transport-msg.tex

```diff
@@ -650,38 +650,41 @@ \subsubsection{Device Notifications}
 \end{itemize}
 
-These messages enable asynchronous updates from the device, reducing driver
-polling overhead and improving performance.
-
+These messages enable asynchronous updates from the device.
 
 \subsubsection{Configuration Changes During Operation}
 \label{sec:Virtio Transport Options / Virtio Over Messages / Device Operation / ConfigUpdates}
 
-The driver \emph{MAY} issue new \msgref{SET_CONFIG} or \msgref{SET_FEATURES}
-messages after initialization if device functionality or features need to be
-changed at runtime. The device \emph{MUST} respond with updated generation counts
-or feature bits if the request succeeds, or \msgref{ERROR} if the request fails
-(e.g., invalid generation count or unsupported features). The driver \emph{SHOULD}
-be prepared to handle partial acceptance of requested features if the device
-cannot enable them all simultaneously.
+The driver \emph{MAY} issue new \msgref{SET_CONFIG}
+messages after initialization if device configuration needs to be
+changed at runtime. The device \emph{MAY} also make configuration changes at
+runtime and \emph{MUST} signal those changes with \msgref{EVENT_CONFIG}
+messages.
+
+\subsubsection{Virtqueue Changes During Operation}
+\label{sec:Virtio Transport Options / Virtio Over Messages / Device Operation / VQueueUpdates}
+
+The driver \emph{MAY} issue new \msgref{SET_VQUEUE} for virtqueues that have not
+yet be setup.  If the VIRTIO_F_RING_RESET feature has been negotiated,
+individual virtqueues can be reset and then optionally re-configured.
 
 \subsubsection{Device Reset and Shutdown}
 \label{sec:Virtio Transport Options / Virtio Over Messages / Device Operation / Reset}
 
 The driver \emph{MAY} reset or shut down a device at any time by writing a status
 of 0 via \msgref{SET_DEVICE_STATUS}. This forces the device to discard its state
 and any pending operations on virtqueues. Once a device is reset, the driver
-\emph{MUST} reinitialize it (see
+\emph{MAY} reinitialize the device (see
 \ref{sec:Virtio Transport Options / Virtio Over Messages / Device Initialization})
-before resuming normal operation.
+if it wishes to try and use the device again.
 
-\paragraph{Backend-Initiated Reset}
-In some circumstances, the device backend \emph{MAY} also trigger a reset if it
+\paragraph{Device Side Initiated Reset}
+In some circumstances, the device \emph{MAY} also trigger a reset if it
 encounters an unrecoverable error. This can be signaled to the driver by sending
-an \msgref{EVENT_CONFIG} with a \emph{device status} indicating an error state,
-or through a higher-level bus reset (\busref{STATUS} if implemented). The driver
-\emph{SHOULD} treat this situation as if \msgref{SET_DEVICE_STATUS}(0) was
-received and re-initialize the device accordingly.
+an \msgref{EVENT_CONFIG} with a \emph{device status} with the DEVICE_NEEDS_RESET
+bit set.  The driver \emph{SHOULD} reset the device after this using a
+\msgref{SET_DEVICE_STATUS}(0) message.  The driver \emph{MAY} reinitialize the
```

Comment:
SET_DEVICE_STATUS(0) should be changed to be more coherent to the format used before:
by writing a status of 0 via VIRTIO_MSG_SET_DEVICE_STATUS.